File: 프로그래머스/lv2/42587. 프로세스/프로세스.java
import java.util.*;
class Solution {
    public int solution(int[] priorities, int location) {
        
        Queue<Custom> q = new LinkedList<>();
        for(int i=0; i<priorities.length; i++){
            q.offer(new Custom(i,priorities[i]));
        }
        
        List<Integer> list = new ArrayList<>();
        for(int pri: priorities)
            list.add(pri);   
        Collections.sort(list, Collections.reverseOrder());
        int answer = 0;
        while(!q.isEmpty()){
            Custom now = q.poll();
            if(now.pri == list.get(0) && now.ord == location){
                answer ++;
                break;
            }else if(now.pri == list.get(0)){
                answer++;
                list.remove(0);
            }
            else{
                q.offer(now);
            }
            
        }
        
        return answer;
    }
    
    class Custom{
        int ord;
        int pri;
        Custom(int ord,int pri){
            this.ord=ord; this.pri=pri;
        }
    }
}

File: 프로그래머스/lv2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    heap = [(val,idx) for idx,val in enumerate(priorities)]
    silhang = 0
    while heap:
        Priority = max(priorities)
        deagi = heap.pop(0)
        if deagi[0] == Priority:
            silhang+=1
            if deagi[1] == location:
                return silhang
            priorities.remove(Priority)
            continue
        else:
            heap.append(deagi)
            
    